north/src/runtime/linux/device_mapper.rs

```diff
@@ -174,13 +176,18 @@ impl Dm {
     /// Create a DM device. It starts out in a "suspended" state.
     ///
     /// Valid flags: DM_READONLY, DM_PERSISTENT_DEV
-    pub async fn device_create(&self, name: &str, options: &DmOptions) -> Result<DeviceInfo> {
+    pub async fn device_create(
+        &self,
+        name: &str,
+        options: &DmOptions,
+    ) -> Result<DeviceInfo, DeviceMapperError> {
         let mut hdr = options.to_ioctl_hdr(None, DmFlags::DM_READONLY | DmFlags::DM_PERSISTENT_DEV);
 
         Self::hdr_set_name(&mut hdr, name);
 
         self.do_ioctl(DM_DEV_CREATE_CMD as u8, &mut hdr, None)
-            .await?;
+            .await
+            .map_err(|_| DeviceMapperError::CreateDeviceFailed);
```

Comment:
This elides the original io::Error. `CreateDeviceFailed` should wrap a `io::Error`